### PR TITLE
add example migration and migrations folder

### PR DESCRIPTION
**Pull request by CJ-Wright**
Checklist
* [ ] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by scopatz** [comment]
Is this safe to merge?

File: recipe/migrations/example.exyaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+__migrator: kind: version
```
Comment (isuruf) [inline recipe/migrations/example.exyaml]: Should this file be in `recipe`? This doesn't belong here IMO.

I think this should go in `.ci_support`.

**Comment by CJ-Wright** [comment]
This is pending merge of the cfep-9 spec

File: recipe/migrations/example.exyaml (hunk 1 above)
Comment (CJ-Wright) [reply recipe/migrations/example.exyaml]: @mariusvniekerk raised the issue that rerenders might do bad things to this. He might have more info on this though.

Comment (isuruf) [reply recipe/migrations/example.exyaml]: That can be fixed easily by not deleting this folder.

Comment (CJ-Wright) [reply recipe/migrations/example.exyaml]: Can you put in a PR/edits into the conda-smithy PR to change where it looks for the migrations?

**Comment by CJ-Wright** [comment]
cfep9 has passed, can someone please merge this when the get the chance. @conda-forge/conda-forge-pinning

**Pull request merged**